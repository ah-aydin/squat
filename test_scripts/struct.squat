struct Person {
    string name;
    int age;
    int height;
}

func func1() {
    println("This is func 1");
}

func add2Persons(Person person1, Person person2) Person {
    return Person(person1.name + person2.name, person1.age + person2.age, person1.height + person2.height);
}

func main() {
    Person person = Person("gigel", 20, 184);
    println(Person);
    println(person);
    println(person.name);
    print("The sum of the age and height is ");
    var the_sum = person.age + person.height;
    println(person.age + person.height);
    println(the_sum);
    println(type(the_sum));

    Person person2 = Person("kebab chef", 21, 140);
    println("Both people have in total " + to_str(person.age + person2.age) + " years lived");

    int height = Person("meh", 10, 170).height;
    println(height);

    person = Person("reassign person", 123, 123);
    println(person.name);

    Person person1toAdd = Person("person1", 25, 175);
    Person person2toAdd = Person("person2", 20, 180);

    Person addedPerson = add2Persons(person1toAdd, person2toAdd);
    println(addedPerson);
}
