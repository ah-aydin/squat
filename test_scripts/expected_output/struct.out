Compiling and running file: test_scripts/struct.squat
<class Person>
<instance of Person [String("gigel"), Int(20), Int(184)]>
gigel
The sum of the age and height is 204
204
<type Int>
Both people have in total 41 years lived
170
reassign person
<instance of Person [String("person1person2"), Int(45), Int(355)]>
